packages/contracts/contracts/optimistic-ethereum/chain/CanonicalTransactionChain.sol

```diff
@@ -178,30 +183,61 @@ contract CanonicalTransactionChain is ContractResolver {
         );
 
         require(
-            _timestamp <= now,
-            "Cannot submit a batch with a timestamp in the future"
+            _blocknumber + forceInclusionPeriodBlocks > block.number,
+            "Cannot submit a batch with a blocknumber older than the sequencer inclusion period"
         );
 
         require(
-            l1ToL2Queue.isEmpty() || _timestamp <= l1ToL2Queue.peekTimestamp(),
-            "Must process older L1ToL2Queue batches first to enforce timestamp monotonicity"
+            _timestamp <= now,
+            "Cannot submit a batch with a timestamp in the future"
         );
 
         require(
-            safetyQueue.isEmpty() || _timestamp <= safetyQueue.peekTimestamp(),
-            "Must process older SafetyQueue batches first to enforce timestamp monotonicity"
+            _blocknumber <= block.number,
+            "Cannot submit a batch with a blocknumber in the future"
         );
 
+        if (!l1ToL2Queue.isEmpty()) {
+            require(
+                _timestamp <= l1ToL2Queue.peekTimestamp(),
+                "Must process older L1ToL2Queue batches first to enforce OVM timestamp monotonicity"
+            );
+
+            require(
+                _blocknumber <= l1ToL2Queue.peekBlocknumber(),
+                "Must process older L1ToL2Queue batches first to enforce OVM blocknumber monotonicity"
+            );
+        }
+
+        if (!safetyQueue.isEmpty()) {
+            require(
+                _timestamp <= safetyQueue.peekTimestamp(),
+                "Must process older SafetyQueue batches first to enforce OVM timestamp monotonicity"
+            );
+
+            require(
+                _blocknumber <= safetyQueue.peekBlocknumber(),
+                "Must process older SafetyQueue batches first to enforce OVM blocknumber monotonicity"
+            );
```

Comment:
Can this be consistently be DoS'd by queueing txs with an old block number / timestamp? Could be useful to also have some sort of force inclusion period enforced on the queues.

Reply:
The block number and timestamp of the queues are set as the `block.number` and `block.timestamp` at the time they are rolled up, so not sure what you mean by "old".  Generally this logic is an exact duplicate of the existing timestamp logic, so it shouldn't fundamentally change any ordering with the sequencer--though it does need they mean to track/update the block number alongside the existing timestamp.

In general, people can spam the queues and force the sequencer to break up batches--have some thoughts on this I'll share separately.